src/Lykke.Bil2.RabbitMq/Subscription/Core/EnvelopedMessage.cs

```diff
@@ -64,21 +72,35 @@ public TimeSpan Age
         
         public string RoutingKey { get; }
         
-        
         public void Ack()
         {
+            _telemetryOperation.Telemetry.ResponseCode = nameof(Ack);
+            _telemetryOperation.Telemetry.Success = true;
+            _telemetryOperation.Telemetry.Stop();
+
+            TelemetryClient.TrackRequest(_telemetryOperation.Telemetry);
+
             _consumer.Ack(_deliveryTag);
         }
 
         public void Reject()
         {
+            _telemetryOperation.Telemetry.ResponseCode = nameof(Reject);
+            _telemetryOperation.Telemetry.Success = false;
+            _telemetryOperation.Telemetry.Stop();
+
+            TelemetryClient.TrackRequest(_telemetryOperation.Telemetry);
+
             _consumer.Reject(_deliveryTag);
         }
 
         public EnvelopedMessage WithIncreasedRetryCount()
         {
             var newRetryCount = RetryCount + 1;
             
+            _telemetryOperation.Telemetry.Sequence = newRetryCount.ToString();
```

Comment:
In this case we need to pass existing telemetry operation to the new EnvelopedMessage and do not stop it. It is needed to track operation from the moment of receiving of the message to the moment of its ack/reject.